x/version/abci.go

```diff
@@ -0,0 +1,22 @@
+package version
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	coretypes "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+// EndBlocker will modify the app version if the current height is equal to
+// a predefined height at which the app version should be changed.
```

Comment:
Late comment to this.

Will this still be activated if more than 2/3rds of the validator set didn't swap the binary before the upgrade height?

Reply:
nodes that did upgrade will, and nodes that didn't won't. So if 2/3s don't, then they can continue producing blocks while the nodes that did upgrade will halt.

Reply:
Awesome :+1: